packages/@react-spectrum/list/stories/ListView.stories.tsx

```diff
@@ -1240,3 +1280,864 @@ function ActionBarExample(props?) {
+              <>
+                <Folder />
+                <Text slot="description">{`contains ${item.childNodes.length} dropped item(s)`}</Text>
+              </>
+            }
+          </Item>
+        )}
+      </ListView>
+    </Flex>
+  );
+}
+
+function FinderDropUtilHandlers(props) {
+  let {listViewProps, firstListDnDOptions, secondListDnDOptions} = props;
+  let list1 = useListData({
+    initialItems: folderList1,
+    getKey: (item) => item.identifier
+  });
+  let list2 = useListData({
+    initialItems: folderList2,
+    getKey: (item) => item.identifier
+  });
+
+  let {dndHooks: list1Hooks} = useDnDHooks({
+    acceptedDragTypes: ['directory'],
+    onInsert: async (e) => {
+      action('onInsertList1')(e);
+    },
+    onItemDrop: async (e) => {
+      action('onItemDropList1')(e);
+    },
+    ...firstListDnDOptions
+  });
+
+  let {dndHooks: list2Hooks} = useDnDHooks({
+    onInsert: async (e) => {
+      action('onInsertList2')(e);
+    },
+    onItemDrop: async (e) => {
+      action('onItemDropList2')(e);
+    },
+    acceptedDragTypes: 'all',
+    ...secondListDnDOptions
+  });
+
+  return (
+    <Flex wrap gap="size-300">
+      <ListView
+        aria-label="ListView that accepts directory drops only"
+        selectionMode="multiple"
+        width="size-3600"
+        height="size-3600"
+        items={list1.items}
+        {...listViewProps}
+        dndHooks={list1Hooks}>
+        {(item: any) => (
+          <Item textValue={item.name} key={item.identifier}>
+            <Text>{item.name}</Text>
+            {item.type === 'folder' &&
+              <>
+                <Folder />
+                <Text slot="description">{`contains ${item.childNodes.length} dropped item(s)`}</Text>
+              </>
+            }
+          </Item>
+        )}
+      </ListView>
+      <ListView
+        aria-label="ListView that accepts all drag types"
+        selectionMode="multiple"
+        width="size-3600"
+        height="size-3600"
+        items={list2.items}
+        {...listViewProps}
+        dndHooks={list2Hooks}>
+        {(item: any) => (
+          <Item textValue={item.name} key={item.identifier}>
+            <Text>{item.name}</Text>
+            {item.type === 'folder' &&
+              <>
+                <Folder />
+                <Text slot="description">{`contains ${item.childNodes.length} dropped item(s)`}</Text>
+              </>
+            }
+          </Item>
+        )}
+      </ListView>
+    </Flex>
+  );
+}
+
+export function DragBetweenListsComplex(props) {
+  let {listViewProps, firstListDnDOptions, secondListDnDOptions} = props;
+  let list1 = useListData({
+    initialItems: folderList1,
+    getKey: (item) => item.identifier
+  });
+
+  let list2 = useListData({
+    initialItems: folderList2,
+    getKey: (item) => item.identifier
+  });
+  let acceptedDragTypes = ['file', 'folder', 'text/plain'];
+
+  let itemProcessor = async (items, acceptedDragTypes) => {
```

Comment:
dup of above function?

Reply:
whoops, pulled it out but for got to replace, thanks for catching